src/wallet/wallet2.cpp

```diff
@@ -1608,11 +1611,12 @@ void wallet2::process_new_blockchain_entry(const cryptonote::block& b, const cry
     m_callback->on_new_block(height, b);
 }
 //----------------------------------------------------------------------------------------------------
-void wallet2::get_short_chain_history(std::list<crypto::hash>& ids) const
+void wallet2::get_short_chain_history(std::list<crypto::hash>& ids, uint64_t granularity) const
 {
   size_t i = 0;
   size_t current_multiplier = 1;
-  size_t sz = m_blockchain.size() - m_blockchain.offset();
+  size_t blockchain_size = m_blockchain.size() / granularity * granularity;
+  size_t sz = blockchain_size - m_blockchain.offset();
```

Comment:
Is `blockchain_size` guaranteed to be no less than `m_blockchain.offset()`?

Reply:
Good point.